src/backends/fun.ts

```diff
@@ -0,0 +1,510 @@
+      }, {
+        params: [FLOAT2, FLOAT2],
+        ret: (args) => `vec2.max(vec2.create(), ${args[0]}, ${args[1]})`,
+      }, {
+        params: [FLOAT3, FLOAT3],
+        ret: (args) => `vec3.max(vec3.create(), ${args[0]}, ${args[1]})`,
+      }, {
+        params: [FLOAT4, FLOAT4],
+        ret: (args) => `vec4.max(vec4.create(), ${args[0]}, ${args[1]})`,
+      }, {
+        params: [FLOAT3X3, FLOAT3X3],
+        ret: (args) => `${args[0]}.map((val, idx) => (Math.max(val, ${args[1]}[idx])))`,
+      }, {
+        params: [FLOAT4X4, FLOAT4X4],
+        ret: (args) => `${args[0]}.map((val, idx) => (Math.max(val, ${args[1]}[idx])))`,
+      }, {
+        params: [FLOAT2, FLOAT],
+        ret: (args) => `${args[0]}.map((val) => (Math.max(val, ${args[1]})))`,
+      }, {
+        params: [FLOAT3, FLOAT],
+        ret: (args) => `${args[0]}.map((val) => (Math.max(val, ${args[1]})))`,
+      }, {
+        params: [FLOAT4, FLOAT],
+        ret: (args) => `${args[0]}.map((val) => (Math.max(val, ${args[1]})))`,
+      },
+    ]
+  }, {
+    func: "clamp", paramsList: [
+      {
+        params: [FLOAT, FLOAT, FLOAT],
+        ret: (args) => `Math.min(Math.max(${args[0]}, ${args[1]}), ${args[2]})`,
+      }, {
+        params: [FLOAT2, FLOAT, FLOAT],
+        ret: (args) => `${args[0]}.map((ele) => (Math.min(Math.max(ele, ${args[1]}), ${args[2]})))`,
+      }, {
+        params: [FLOAT3, FLOAT, FLOAT],
+        ret: (args) => `${args[0]}.map((ele) => (Math.min(Math.max(ele, ${args[1]}), ${args[2]})))`,
+      }, {
+        params: [FLOAT4, FLOAT, FLOAT],
+        ret: (args) => `${args[0]}.map((ele) => (Math.min(Math.max(ele, ${args[1]}), ${args[2]})))`,
+      }, {
+        params: [FLOAT2, FLOAT2, FLOAT2],
+        ret: (args) => `vec2.min(vec2.max(${args[0]}, ${args[1]}), ${args[2]})`,
+      }, {
+        params: [FLOAT3, FLOAT3, FLOAT3],
+        ret: (args) => `vec3.min(vec3.max(${args[0]}, ${args[1]}), ${args[2]})`,
+      }, {
+        params: [FLOAT4, FLOAT4, FLOAT4],
+        ret: (args) => `vec4.min(vec4.max(${args[0]}, ${args[1]}), ${args[2]})`,
+      },
+    ]
+  }, {
+    func: "exp2", paramsList: [
+      {
+        params: [FLOAT],
+        ret: (args) => `Math.pow(2.0, ${args[0]})`,
+      },
+    ]
+  }, {
+    func: "cross", paramsList: [
+      {
+        params: [FLOAT3, FLOAT3],
+        ret: (args) => `vec3.cross(vec3.create(), ${args[0]}, ${args[1]})`,
+      },
+    ]
+  }, {
+    func: "mix", paramsList: [
+      {
+        params: [FLOAT, FLOAT, FLOAT],
+        ret: (args) => `${args[0]} * (1.0 - ${args[2]}) + ${args[1]} * ${args[2]}`,
+      }, {
+        params: [FLOAT2, FLOAT2, FLOAT],
+        ret: (args) => `vec2.lerp(vec2.create(), ${args[0]}, ${args[1]}, ${args[2]})`,
+      }, {
+        params: [FLOAT3, FLOAT3, FLOAT],
+        ret: (args) => `vec3.lerp(vec3.create(), ${args[0]}, ${args[1]}, ${args[2]})`,
+      },
+    ]
+  }
+]
+
+export function getFunc(func: string, params: Type[], args: string[]) {
```

Comment:
This all looks good. If I may make two nontechnical suggestions:

* What do you think about the name `webglfun` for this module, instead of just `fun`? Someday, for example, we might also add an `openglfun` version for the native backend.
* Would you mind adding a comment to say what `getFunc` does, and perhaps also one describing the parts of `FuncMapType`? This could help with long-term maintainability.